modules/sagemaker/sagemaker-templates-service-catalog/templates/model_deploy/product_stack.py

```diff
@@ -110,102 +110,144 @@ def __init__(
-                principals=[iam.AnyPrincipal()],
-            )
-        )
-        # Add cross-account access
-        pipeline_artifact_bucket.add_to_resource_policy(
-            iam.PolicyStatement(
-                sid="CrossAccountPermissions",
-                actions=["s3:List*", "s3:Get*", "s3:Put*"],
-                resources=[
-                    pipeline_artifact_bucket.arn_for_objects(key_pattern="*"),
-                    pipeline_artifact_bucket.bucket_arn,
-                ],
-                principals=[
-                    iam.AccountPrincipal(pre_prod_account_id),
-                    iam.AccountPrincipal(prod_account_id),
+                    iam.PolicyStatement(
+                        actions=["ssm:GetParameter"],
+                        resources=[
+                            f"arn:{Aws.PARTITION}:ssm:{dev_region}:{dev_account_id}:parameter/*",
+                            f"arn:{Aws.PARTITION}:ssm:{pre_prod_region}:{pre_prod_account_id}:parameter/*",
+                            f"arn:{Aws.PARTITION}:ssm:{prod_region}:{prod_account_id}:parameter/*",
+                        ],
+                    )
                 ],
             )
         )
 
-        DeployPipelineConstruct(
+        # Create a CloudWatch Event Rule to trigger the CodeBuild project
+        build_project_event_rule = events.Rule(
             self,
-            "deploy",
-            project_name=sagemaker_project_name,
-            project_id=sagemaker_project_id,
-            model_bucket=model_bucket,
-            pipeline_artifact_bucket=pipeline_artifact_bucket,
-            model_package_group_name=model_package_group_name,
-            repo_asset=deploy_app_asset,
-            dev_vpc_id=dev_vpc_id,
-            dev_subnet_ids=dev_subnet_ids,
-            dev_security_group_ids=dev_security_group_ids,
-            pre_prod_account_id=pre_prod_account_id,
-            pre_prod_region=pre_prod_region,
-            pre_prod_vpc_id=pre_prod_vpc_id,
-            pre_prod_subnet_ids=pre_prod_subnet_ids,
-            pre_prod_security_group_ids=pre_prod_security_group_ids,
-            prod_account_id=prod_account_id,
-            prod_region=prod_region,
-            prod_vpc_id=prod_vpc_id,
-            prod_subnet_ids=prod_subnet_ids,
-            prod_security_group_ids=prod_security_group_ids,
+            "BuildProjectEventRule",
+            description="Trigger the CodeBuild project when it's created",
+            event_pattern=events.EventPattern(
+                source=["aws.codebuild"],
+                detail_type=["CodeBuild Build Project State Change"],
+                detail={
+                    "project-name": [CodePipelineDeployProjectName],
+                    "build-status": ["SUCCEEDED"],
+                },
+            ),
         )
+        # Add the CodeBuild project as a target for the CloudWatch Event Rule
+        build_project_event_rule.add_target(targets.CodeBuildProject(project))
```

Comment:
I'm not sure what this is doing. The codebuild job is both the event and the target? So if the job succeeds, it triggers again?

Also, when I deployed the catalog product, this codebuild job didn't trigger automatically. I had to go into CodeBuild and trigger it in order to create the pipeline.